sim-lib/src/lib.rs

```diff
@@ -573,19 +578,25 @@ impl Simulation {
         // Note: when we allow configuring both defined and random activity, this will no longer be an if/else, we'll
         // just populate with each type as configured.
         if !self.activity.is_empty() {
-            for description in self.activity.iter() {
+            for (index, description) in self.activity.iter().enumerate() {
                 let activity_generator = DefinedPaymentActivity::new(
                     description.destination.clone(),
                     Duration::from_secs(description.interval_secs.into()),
                     description.amount_msat,
                 );
 
+                let activity_name = match &description.activity_name {
+                    Some(name) => name.clone(),
+                    None => index.to_string(),
```

Comment:
perhaps "index {}" so that users understand where the number comes from

Reply:
Sure! I'll fix that alongside the other changes and update the PR.